src/pages/status.tsx

```diff
@@ -190,6 +197,114 @@ const Status = () => {
     }
   }
 
+  // This is just a first draft of this logic.
+  // We'll ultimately have to come up with logic based
+  // on both the status + the dates (in case they get out of sync).
+  const getTimelineEntries = (
+    response: CheckStatusApiResponse,
+  ): TimelineEntryData[] => {
+    let entries: TimelineEntryData[] = []
+
+    if (response.receivedDate) {
+      entries.push({
+        status: 'done',
+        date: response.receivedDate,
+        step: t('timeline:received'),
+      })
+    }
+
+    if (response.reviewedDate) {
+      entries.push({
+        status: 'done',
+        date: response.reviewedDate,
+        step: t('timeline:review-done'),
+      })
+    } else {
+      entries.push({ status: 'current', step: t('timeline:review-current') })
+    }
+
+    // Documents returned is a special case.
+    // We don't have any timeline entries after this if it exists.
+    if (response.documentsReturnedDate) {
+      entries.push({
+        status: 'done',
+        date: response.documentsReturnedDate,
+        step: t('timeline:documents-returned'),
+      })
+      return entries
+    }
+
+    if (response.printedDate) {
+      entries.push({
+        status: 'done',
+        date: response.printedDate,
+        step: t('timeline:print-done'),
+      })
+    } else if (response.reviewedDate) {
+      entries.push({ status: 'current', step: t('timeline:print-current') })
+    } else {
+      entries.push({ status: 'future', step: t('timeline:print-future') })
+    }
+
+    if (response.submissionType === 'mail') {
+      if (response.mailedDate) {
+        entries.push({ status: 'done', step: t('timeline:mail-future') })
+      } else if (response.printedDate) {
+        entries.push({ status: 'current', step: t('timeline:mail-current') })
+      } else {
+        entries.push({ status: 'future', step: t('timeline:mail-future') })
+      }
+    } else {
+      if (response.pickUpReadyDate) {
+        entries.push({ status: 'done', step: t('timeline:pickup-done') })
+      } else if (response.printedDate) {
+        entries.push({ status: 'current', step: t('timeline:pickup-future') })
+      } else {
+        entries.push({ status: 'future', step: t('timeline:pickup-future') })
+      }
+    }
+
+    return entries
+  }
+
+  const getStatusComponent = (response: CheckStatusApiResponse | null) => {
+    if (response !== null) {
+      let timelineData = getTimelineEntries(response)
+
+      switch (response.status) {
```

Comment:
This is more of a nitpick, so feel free to ignore..

I usually prefer wrapping each case block in curly braces for better maintainability, which helps when future changes might involve adding more logic to a block. For example:

``` typescript
switch (thing) {
  case 'Case1': {
    return doSomething();
  }

  case 'Case2': {
    throw new ApplicationError('barf!');
  }
}
```

This helps avoid subtle bugs that could be introduced when adding new code to a case block.

Reply:
True, but I pulled this from the component I deleted, so it was more or less existing code. And I don't think it'll ever be more than just a simple mapping of enum => component. So won't bother adding any bloat at this point.